ai/core/src/unitycatalog/ai/core/utils/function_processing_utils.py

```diff
@@ -391,3 +391,37 @@ def execute_retriever(parameters):
             f"Skipping tracing {function_info.full_name} as a retriever because of the following error:\n {e}"
         )
         return _execute_uc_function(function_info, parameters, **kwargs)
+
+
+def process_function_parameter_defaults(
+    function_info: "FunctionInfo", parameters: Optional[dict[str, Any]] = None
+) -> dict[str, Any]:
+    """
+    Handle default values for input parameters.
+
+    Args:
+        function_info: The FunctionInfo object containing the function metadata.
+        parameters: The parameters to handle.
+
+    Returns:
+        The updated parameters with default values filled in.
+    """
+    defaults = {}
+    if function_info.input_params and function_info.input_params.parameters:
+        for param in function_info.input_params.parameters:
+            if param.parameter_default is not None:
+                default_str = param.parameter_default.strip()
+                if default_str.upper() == "NULL":
+                    defaults[param.name] = None
+                elif default_str.upper() == "TRUE":
+                    defaults[param.name] = True
+                elif default_str.upper() == "FALSE":
+                    defaults[param.name] = False
```

Comment:
```suggestion
                if default_str.upper() in default_string_val_mapping:
                    defaults[param.name] = default_string_val_mapping[default_str.upper()]
```
Could we create a dictionary for the mapping?

Reply:
Good idea. This conditional chain is a bit annoying to read :)